templates/common/baremetal/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -49,6 +47,37 @@ contents:
             show \
             "{{.Infra.Status.PlatformStatus.BareMetal.APIServerInternalIP}}" \
             "{{.Infra.Status.PlatformStatus.BareMetal.IngressIP}}")
+        # check if we got DHCPv6 hostname. The value would usually be in
+        # DHCP6_FQDN_FQDN but NM internal client does not parse the option
+        # Only if it is IPv4
+        if grep -q ":" "$NAMESERVER_IP"; then
+            LEASE_FILE=$(mktemp)
+            const CONT_NAME="dhcpv6-hostname-workaround"
+            /usr/bin/podman run --rm -d --name "$CONT_NAME" \
+                --authfile /var/lib/kubelet/config.json \
+                --net=host \
+                --entrypoint /usr/sbin/dhclient \
+                -v "${LEASE_FILE}:/tmp/workaround.lease:Z" \
+                {{ .Images.baremetalRuntimeCfgImage }} \
+                -d  -sf /bin/true -lf /tmp/workaround.lease -6 -v "$IFACE"
+            DHCPv6_DHCLIENT_TIMEOUT=15
+            while : ; do
+                mapfile -t NAME_DATA < <(awk '/(host|domain)-name / {print substr($3, 2, length($3) - 3)}' "$LEASE_FILE")
+                if [[ ${#NAME_DATA[@]} -eq 2 ]]; then
+                    hostnamectl set-hostname --static --transient "${NAME_DATA[0]}.${NAME_DATA[1]}"
+                    break
+                else
+                    if [[ DHCPv6_DHCLIENT_TIMEOUT -gt 0 ]]; then
+                        sleep 0.5
+                        DHCPv6_DHCLIENT_TIMEOUT=$((DHCPv6_DHCLIENT_TIMEOUT - 1))
+                    else
+                        break
+                    fi
+                fi
+            done
+            podman rm -f "$CONT_NAME"
```

Comment:
I wonder if we should put this in a trap so it will still get run even if an earlier command fails. set -e means we could potentially exit the script before this point.

Reply:
That's a very good point.